README.md

```diff
@@ -1,2 +1,101 @@
+Podczas tworzenia nowego projektu użytkownik może wybrać jeden z poniższych szablonów:
+<ul>
+    <li>Soneta Addon Project – to właściwy projekt logiki dodatku</li>
+    <li>Soneta Addon Config – w jego skład wchodzą pliki konfiguracyjne wymagane do prawidłowego działania projektów</li>
+    <li>Soneta Addon Project Tests– projekt zawierający testy</li>
+    <li>Soneta Addon Project UI – projekt zawierający elementy interfejsu użytkownika dla dodatku</li>
+    <li>Soneta Addon- jest to kompletny projekt dodatku zawierający wszystkie powyższe szablony.</li>
+</ul>
+<img src="Soneta.Platform.Developer\documentation\pictures\vs_projects_list.jpg">
+
+# Instalacja
+Dodatek Soneta Platform Developer może zostać zainstalowany za pomocą konsoli używając polecenia:
+```
+„dotnet new -i Soneta.Platform.Developer”.
+```
+Listę zainstalowanych dodatków możemy podejrzeć za pomocą polecenia:
+```
+dotnet new -l
+```
+Dodatek może być także zainstalowany z poziomu **Visual Studio**. Robimy to poprzez menu **Extensions --> Manage Extensions**, a następnie **wyszukujemy Soneta Platform Developer**.  Pobieramy i instalujemy nasz dodatek, w tym celu należy ponownie uruchomić Visual Studio. Po poprawnym zainstalowaniu naszego dodatku będziemy mieli możliwość stworzenia nowego projektu według zainstalowanych szablonów.
+# Tworzenie nowego dodatku przyuzyciu Visual Studio
+Tworzenie nowego dodatku w Visual Studio jest bardzo proste i sprowadza się jedynie do wybrania odpowiedniego szablonu. W celu utworzenie pełnego projektu dodatku zaleca się użycie szablonu **Soneta Addon**.
+# Tworzenie nowego dodatku przy użyciu Visual Studio Code
+W celu wykorzystania Visual Studio Code na początku należy zainstalować wtyczkę „**C# for Visual Studio Code**” oraz dodatek Soneta Platform Developer. Następnie za pomocą konsoli utworzyć nowy projekt dotatku wybierając szablon **soneta-addon** oraz utworzyć nową solucję. W kolejnym kroku **do solucji** należy dodać wszystkie **utworzone projekty** oraz stworzyć domyślne **taski** do **kompilacji i debugowania**. Cały proces został opisany szczegółowo poniżej.
+<br>
+
+Na początku otwieramy **Visual Studio Code** i przechodzimy do zakładki **Extensions**( Ctrl+ Shift +x) i instalujemy wtyczkę „C# for Visual Studio Code”.<br>
+<img src="Soneta.Platform.Developer\documentation\pictures\vsc_extension.jpg">
+
+Następnie otworzymy **terminal** w Visual Studio Code (View--> Terminal)  i **instalujemy dodatek Soneta Platform Developer**.  Jest do dodatek , który zawiera nowe szablony. W celu instalacji dodatku wywołujemy w konsoli komendę: **„dotnet new -i Soneta.Platform.Developer”**.
+Za pomocą plecenia „**dotnet new -l**” możemy podejrzeć listę wszystkich zainstalowanych szablonów.
+<br>
+
+Następnie należy dodać do naszego obszaru roboczego folder w którym umieszczony będzie nasz dodatek (File-> Add Folder To Workspace…). Mój folder będzie nazywał się „MyExtension”.
```

Comment:
zaleceane jest nie mieszanie form gamatycznych - ale to mozesz olac
czyli jak wczesniej piszemy otwieramy - to tutaj powinno być dodajemy.
Ja preferuję formę należy dodać, należy otworzyć ( bezosobową)
przecinek przed w którym